lib/r10k/util/purgeable.rb

```diff
@@ -56,6 +56,29 @@ def stale_contents(recurse, exclusions, whitelist)
 
           !!exclusion_match || !!whitelist_match
         end
+
+        children.flat_map do |child|
+          if File.directory?(child) && recurse
+            potentially_purgeable(child, exclusions, whitelist, recurse, fn_match_opts)
+          else
+            child.to_s
+          end
+        end.compact
```

Comment:
From what I can tell, we should never get nils in here, but maybe I'm missing something?

Reply:
Yeah, good catch. There was a version that did and needed this but I updated it without removing the additional compaction.